### PR TITLE
Misc x64 fixes

### PR DESCRIPTION
**Pull request by tkreuzer**
* [RTL/x64] Properly handle UNW_FLAG_CHAININFO
* [NETAPI32] Zero out lpNameBuffer in NetGetJoinInformation
* [NTDLL_APITEST] Add test for some user mode exceptions
* [NTOS:KE/x64] Handle user faults in KiGeneralProtectionFaultHandler

### CODE REVIEW

File: dll/win32/netapi32/wksta_new.c [hunk 1]
```diff
@@ -327,6 +327,7 @@ NetGetJoinInformation(
     if (lpNameBuffer == NULL || BufferType == NULL)
         return ERROR_INVALID_PARAMETER;
 
+    *lpNameBuffer = NULL;
```
Comment (SergeGautherie) [inline dll/win32/netapi32/wksta_new.c:330]: Maybe add a HACK comment, as `NetpGetJoinInformation()` already has `*NameBuffer = NULL;`.
https://git.reactos.org/?p=reactos.git;a=blob;f=base/services/wkssvc/domain.c;hb=e8277e51e2ec9c13721b68cd8062b3b7d933d255#l69

Comment (ThFabba) [reply dll/win32/netapi32/wksta_new.c:330]: Should the fix be to remove `in` from https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/reactos/idl/wkssvc.idl;hb=e8277e51e2ec9c13721b68cd8062b3b7d933d255#l536 ? Or is this somehow unsupported by widl/rpcrt4?

File: modules/rostests/apitests/ntdll/UserModeException.c [hunk 2]
```diff
@@ -0,0 +1,136 @@
+/*
+* PROJECT:         ReactOS api tests
+* LICENSE:         GPL - See COPYING in the top level directory
+* PURPOSE:         Test for stack overflow
```
Comment (ThFabba) [inline modules/rostests/apitests/ntdll/UserModeException.c:4]: Copypasta

File: modules/rostests/apitests/ntdll/UserModeException.c [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        Status = GetExceptionCode();
+    }
+    _SEH2_END;
+
+    ok_ntstatus(Status, STATUS_ILLEGAL_INSTRUCTION);
+    ok_long(ExceptionRecord.ExceptionCode, STATUS_ILLEGAL_INSTRUCTION);
+    ok_long(ExceptionRecord.ExceptionFlags, 0);
+    ok_ptr(ExceptionRecord.ExceptionRecord, NULL);
+    ok(((ULONG_PTR)ExceptionRecord.ExceptionAddress >= (ULONG_PTR)Test_IllegalInstructionSingle) &&
+       ((ULONG_PTR)ExceptionRecord.ExceptionAddress < 0x00007FFFFFFFFFFF),
```
Comment (ThFabba) [inline modules/rostests/apitests/ntdll/UserModeException.c:43]: Should this be an equivalent constant for x86?

File: modules/rostests/apitests/ntdll/amd64/UserModeException.S [hunk 4]
```diff
@@ -0,0 +1,27 @@
+
+#include <asm.inc>
+
+.code64
+
+PUBLIC Illegal_0F_0B
+.PROC Illegal_0F_0B
+    .ENDPROLOG
+    db HEX(06)
```
Comment (ThFabba) [inline modules/rostests/apitests/ntdll/amd64/UserModeException.S:9]: Should this be 0f 0b?

File: dll/win32/netapi32/wksta_new.c (hunk 1 above)
Comment (tkreuzer) [reply dll/win32/netapi32/wksta_new.c:330]: The annotation matches what Windows has: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-wkst/12d7af34-9efc-4f39-ab88-a8d8e7bcbdc2 and https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-wkst/9fdbc753-0397-4236-bbfc-a380f9d23789

File: modules/rostests/apitests/ntdll/UserModeException.c (hunk 3 above)
Comment (tkreuzer) [reply modules/rostests/apitests/ntdll/UserModeException.c]: I improved it to check for the actual address.

File: sdk/lib/rtl/amd64/unwind.c [hunk 5]
```diff
@@ -650,6 +642,16 @@ RtlVirtualUnwind(
...
     }
 
+    /* Check for chained info */
+    if (UnwindInfo->Flags & UNW_FLAG_CHAININFO)
+    {
+        /* See https://docs.microsoft.com/en-us/cpp/build/exception-handling-x64?view=msvc-160#chained-unwind-info-structures */
+        FunctionEntry = (PRUNTIME_FUNCTION)&(UnwindInfo->UnwindCode[(UnwindInfo->CountOfCodes + 1) & ~1]);
+        UnwindInfo = RVA(ImageBase, FunctionEntry->UnwindData);
+        i = 0;
+        goto RepeatChainedInfo;
```
Comment (learn-more) [inline sdk/lib/rtl/amd64/unwind.c:652]: In theory this can lead to infinite loops?
If one entry points to itself and has this flag.

Comment (zefklop) [reply sdk/lib/rtl/amd64/unwind.c:652]: Well, broken binaries lead to broken behaviour. Is that a problem ?

**Pull request merged**